### PR TITLE
Ext Reg fixes: extension generator edits

### PR DESCRIPTION
**Pull request by shazron**
These changes are against the `release-ext-reg` release branch.

1. change the e2e test file name from `{action}.e2e.js` to `{action}.e2e.test.js` (pattern matching for tests in jest)
2. add the `test` hook to dx-asset-compute for new `aio app test` functionality
3. `e2e` and `test` folders for now relative to where the actions folder is (as a sibling folder). For a --no-extensions application, it will be in the root of the project, for an extension, it is wherever the extension is installed, not the root.
4. `asset-compute` generator: change the test folder bulk copy location (from the root `test` folder to the new location) and remove the installation of the test script in `package.json`
5.  remove `base-app`  file `test/jest.setup.js` from `test` folder to root folder. If there were extensions installed, having this single file in the `test` folder doesn't make sense.

## How Has This Been Tested?

locally via npm link

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: generators/add-action/asset-compute/index.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ class AssetComputeGenerator extends ActionGenerator {
   constructor (args, opts) {
     super(args, opts)
     this.props = {}
+    this.option('action-folder', { type: String })
+
+    this.actionFolder = this.options['action-folder']
```
Comment (moritzraho) [inline generators/add-action/asset-compute/index.js:25]: this should be already defined in the super class https://github.com/adobe/generator-aio-app/blob/e5fcce7741dd70aa7645ec7d3c434b88e4f40b88/lib/ActionGenerator.js#L29

File: generators/add-action/asset-compute/index.js [hunk 2]
```diff
@@ -46,8 +49,11 @@ class AssetComputeGenerator extends ActionGenerator {
         }
       }
     })
+
+    const extFolder = path.dirname(this.actionFolder)
```
Comment (moritzraho) [inline generators/add-action/asset-compute/index.js:53]: That's not always the case unfortunately, as actions can be added post init with a folder structure that has been decided by the user. I think a better approach would be `path.dirname(this.configPath)` - configPath is the path to the config file that defines the extension (retrievable from super class)

File: generators/extensions/dx-asset-compute-worker-1/index.js [hunk 3]
```diff
@@ -101,20 +101,14 @@ class DxAssetComputeWorker1 extends Generator {
       'hooks',
       // value
       {
-        'post-app-run': 'adobe-asset-compute devtool'
+        'post-app-run': 'adobe-asset-compute devtool',
+        test: 'adobe-asset-compute test-worker'
```
Comment (moritzraho) [inline generators/extensions/dx-asset-compute-worker-1/index.js:105]: nice ! should that new hook be documented in https://www.adobe.io/apis/experienceplatform/project-firefly/docs.html#!AdobeDocs/project-firefly/master/guides/app-hooks.md

File: generators/add-action/asset-compute/index.js (hunk 2 above)
Comment (moritzraho) [reply generators/add-action/asset-compute/index.js:53]: (ext config can redefine `actions` config path to point to another action folder)

**Review by moritzraho — COMMENTED** [review]
few comments on some details. Otherwise lgtm and +1 for the idea for self containing test folders for extensions

File: generators/extensions/dx-asset-compute-worker-1/index.js (hunk 3 above)
Comment (shazron) [reply generators/extensions/dx-asset-compute-worker-1/index.js:105]: > nice ! should that new hook be documented in https://www.adobe.io/apis/experienceplatform/project-firefly/docs.html#!AdobeDocs/project-firefly/master/guides/app-hooks.md

added issue for tracking https://github.com/adobe/aio-cli-plugin-app/issues/425

**Comment by shazron** [comment]
Tested this with https://github.com/adobe/aio-cli-plugin-app/pull/418 and gtg

**Comment by moritzraho** [comment]
Nice feel free to merge :)

**Pull request merged**